content/exporters/custom-exporter/Java/Metrics.md

```diff
@@ -0,0 +1,312 @@
+---
+title: "Metrics exporter"
+draft: false
+weight: 3
+aliases: [/custom_exporter/java/stats, /guides/custom_exporter/java/stats]
+---
+
+- [Introduction](#introduction)
+- [Implementation](#implementation)
+- [Runnable example](#runnable-example)
+- [References](#references)
+
+## Introduction
+
+A metrics exporter must extend the abstract class [MetricExporter](https://github.com/census-instrumentation/opencensus-java/blob/master/exporters/metrics/util/src/main/java/io/opencensus/exporter/metrics/util/MetricExporter.java) implementing the `export` method:
```

Comment:
This seems to be a metrics exporter but below uses stats. @bogdandrutu @songy23 is the MetricsExporter our bootstrapping mechanism to go from stats to producing metrics everywhere?

Reply:
Emmanuel, are you asking about the terminology 'stats' vs 'metrics'? I see both used in the docs and would appreciate guidance on the most suitable term to use. The [Go custom exporter example](https://opencensus.io/exporters/custom-exporter/go/metrics/) uses 'metrics exporter'

For the interface implemented, this is the same as the Prometheus and Stackdriver exporters in implementing MetricsExporter:
https://github.com/census-instrumentation/opencensus-java/blob/master/exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java#L95
and
https://github.com/census-instrumentation/opencensus-java/blob/master/exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/CreateTimeSeriesExporter.java#L40

I think that we are missing MetricExporter into the [OpenCensus JavaDoc](https://static.javadoc.io/io.opencensus/opencensus-api/0.19.2/index.html). I thought that there might be a separate bundle for [Java Metrics Exporter Util](https://github.com/census-instrumentation/opencensus-java/tree/master/exporters/metrics/util) but I can't find it.

Reply:
> is the MetricsExporter our bootstrapping mechanism to go from stats to producing metrics everywhere?

Correct. `MetricExporter` is the newly-added abstract class for metrics exporting. OpenCensus library will export metrics to any subclass that extends `MetricExporter`.

> the terminology 'stats' vs 'metrics'?

Please use `metrics` here. `stats` is one of ways that `metrics` get created, and in terms of exporting, we always export `metrics`.

> I thought that there might be a separate bundle for [Java Metrics Exporter Util](https://github.com/census-instrumentation/opencensus-java/tree/master/exporters/metrics/util) but I can't find it.

Javadoc is at https://www.javadoc.io/doc/io.opencensus/opencensus-exporter-metrics-util/.

Reply:
Thanks. I added a link to the JavaDoc with a new push.

Reply:
>Emmanuel, are you asking about the terminology 'stats' vs 'metrics'? I see both used in the docs and would appreciate guidance on the most suitable term to use.

Stats are the raw measurements that are produced e.g. from `measure.M(discreteValute)` and when aggregations and tags are applied by views, those become Metrics; Metrics are also produced by already aggregated values that are collected from other systems.